File: crossbuy_mobile/lib/models/employee.dart
class Employee {
  final int id;
  final String? firstName;
  final String? lastName;
  final String? fullName;
  final String? fullNameEn;
  final String? email;
  final String? phoneNumber;
  final String? address;
  final String? gender;
  final String? maritalStatus;
  final String? profileImage;
  final DateTime? dateOfBirth;
  final DateTime? dateOfJoining;
  final String? jobTitleAr;
  final String? jobTitleEn;
  final String? companyAr;
  final String? companyEn;
  final String? branchAr;
  final String? branchEn;
  final bool isActive;
  final String? employmentType;
  final String? departmentAr;
  final String? departmentEn;

  Employee({
    required this.id,
    this.firstName,
    this.lastName,
    this.fullName,
    this.fullNameEn,
    this.email,
    this.phoneNumber,
    this.address,
    this.gender,
    this.maritalStatus,
    this.profileImage,
    this.dateOfBirth,
    this.dateOfJoining,
    this.jobTitleAr,
    this.jobTitleEn,
    this.companyAr,
    this.companyEn,
    this.branchAr,
    this.branchEn,
    this.isActive = true,
    this.employmentType,
    this.departmentAr,
    this.departmentEn,
  });

  static DateTime? _date(dynamic v) =>
      v == null ? null : DateTime.tryParse(v.toString());

  factory Employee.fromJson(Map<String, dynamic> j) => Employee(
        id: j['id'] ?? j['ID'] ?? 0,
        firstName: j['firstName'],
        lastName: j['lastName'],
        fullName: j['fullName'],
        fullNameEn: j['fullNameEn'],
        email: j['email'],
        phoneNumber: j['phoneNumber'],
        address: j['address'],
        gender: j['gender'],
        maritalStatus: j['maritalStatus'],
        profileImage: j['profileImage'],
        dateOfBirth: _date(j['dateOfBirth']),
        dateOfJoining: _date(j['dateOfJoining']),
        jobTitleAr: j['jobTitleAr'],
        jobTitleEn: j['jobTitleEn'],
        companyAr: j['companyAr'],
        companyEn: j['companyEn'],
        branchAr: j['branchAr'],
        branchEn: j['branchEn'],
        isActive: j['isActive'] ?? true,
        employmentType: j['employmentType'],
        departmentAr: j['departmentAr'],
        departmentEn: j['departmentEn'],
      );
}

File: CrossBuy/Models/Context/Admin/Employee.cs
using static Microsoft.EntityFrameworkCore.DbLoggerCategory.Database;
using System.Diagnostics.Metrics;
using Microsoft.AspNetCore.Identity;
using System.ComponentModel.DataAnnotations;

namespace CrossBuy.Models.Context.Admin
{
	public class Employee : BaseEntity
	{
		[Key]
		public int ID { get; set; } // معرّف الموظف

		public string FirstName { get; set; } // الاسم الأول
		public string LastName { get; set; } // الاسم الأخير
		public string FullName { get; set; } // الاسم الكامل (عربي)
		public string? FullNameEn { get; set; } // الاسم الكامل (إنجليزي)

		public string Address { get; set; } // العنوان
		public string PhoneNumber { get; set; } // رقم الهاتف
		public string Email { get; set; } // البريد الإلكتروني

		public int? CountryID { get; set; } // معرّف الدولة
		public CountriesLookup Country { get; set; } // الربط مع جدول الدول

		public int JobTitleID { get; set; } // معرّف المسمي الوظيفي
		public JobTitle JobTitle { get; set; } // الربط مع جدول المسميات الوظيفية

		public int? BranchID { get; set; } // معرّف الفرع
		public Branch Branch { get; set; } // الربط مع جدول الفروع

		public int EmpCompanyID { get; set; } // معرّف الشركة
		public Companies Company { get; set; } // الربط مع جدول الشركات

		public string ProfileImage { get; set; } // صورة الموظف

		public DateTime DateOfBirth { get; set; } // تاريخ الميلاد
		public string Gender { get; set; } // الجنس
		public string MaritalStatus { get; set; } // الحالة الاجتماعية
		public DateTime DateOfJoining { get; set; } // تاريخ الانضمام
		public bool IsActive { get; set; } // حالة النشاط

		public int? DepartmentID { get; set; }      // → Hierarchical (عقدة الإدارة)
		public string? EmploymentType { get; set; } // نوع التوظيف (دوام كامل/جزئي...)

		public decimal? ManufHourlyRate { get; set; } // سعر ساعة التصنيع الصريح (يتجاوز اشتقاق الراتب)

		public string UserId { get; set; }
		public Users User { get; set; }

		public ICollection<PolicyAssignments> policyAssignments { get; set; }

	}


}
